ui/app/AppLayouts/Communities/popups/CreateChannelPopup.qml

```diff
@@ -43,24 +46,129 @@ StatusDialog {
+            if (error) {
+                creatingError.text = error.error
+                creatingError.open()
+            }
+        }
     }
 
-    title: qsTr("New channel")
-    padding: 0
+    stackTitle: isDiscordImport ? qsTr("New Channel With Imported Chat History") :
+                                  isEdit ? qsTr("Edit #%1").arg(root.channelName)
+                                         : qsTr("New channel")
+
+    nextButton: StatusButton {
+        objectName: "createChannelNextBtn"
+        font.weight: Font.Medium
+        text: typeof currentItem.nextButtonText !== "undefined" ? currentItem.nextButtonText : qsTr("Import chat history")
+        enabled: typeof(currentItem.canGoNext) == "undefined" || currentItem.canGoNext
+        loading: root.communitiesStore.discordDataExtractionInProgress
+        onClicked: {
+            let nextAction = currentItem.nextAction
+            if (typeof(nextAction) == "function") {
+                return nextAction()
+            }
+            root.currentIndex++
+        }
+    }
+
+    finishButton: StatusButton {
+        objectName: "createOrEditCommunityChannelBtn"
+        font.weight: Font.Medium
+        text: isDiscordImport ? qsTr("Import chat history") : isEdit ? qsTr("Save changes") : qsTr("Create channel")
+        enabled: typeof(currentItem.canGoNext) == "undefined" || currentItem.canGoNext
+        onClicked: {
+            let nextAction = currentItem.nextAction
+            if (typeof (nextAction) == "function") {
+                return nextAction()
+            }
+            if (!root.isDiscordImport) {
+                if (!d.isFormValid()) {
+                    scrollView.scrollBackUp()
+                    return
+                }
+                let emoji = StatusQUtils.Emoji.deparse(nameInput.input.asset.emoji)
+
+                if (!isEdit) {
+                    root.createCommunityChannel(StatusQUtils.Utils.filterXSS(nameInput.input.text),
+                                                StatusQUtils.Utils.filterXSS(descriptionTextArea.text),
+                                                emoji,
+                                                colorDialog.color.toString().toUpperCase(),
+                                                root.categoryId)
+                } else {
+                    root.editCommunityChannel(StatusQUtils.Utils.filterXSS(nameInput.input.text),
+                                              StatusQUtils.Utils.filterXSS(descriptionTextArea.text),
+                                              emoji,
+                                              colorDialog.color.toString().toUpperCase(),
+                                              root.categoryId)
+                }
+                // TODO Open the channel once we have designs for it
+                root.close()
+            }
+        }
+    }
+
+    readonly property var clearFilesButton: StatusButton {
```

Comment:
```suggestion
    readonly property StatusButton clearFilesButton: StatusButton {
```
or maybe no property at all, just regular object referred by id? Is it necessary to expose it?

Reply:
It's needed for the footer buttons (`rightButtons` property)